File: presentation/src/main/java/homepunk/github/com/presentation/common/dagger/AppComponent.java
package homepunk.github.com.presentation.common.dagger;

import javax.inject.Singleton;

import dagger.Component;
import homepunk.github.com.data.common.dagger.DataComponent;
import homepunk.github.com.data.common.dagger.DataModule;
import homepunk.github.com.presentation.feature.main.home.HomeFragment;

@Component(modules = {AppModule.class})
@Singleton
public interface AppComponent {
    void inject(HomeFragment fragment);
    DataComponent newDataComponent(DataModule module);
}

File: presentation/src/main/java/homepunk/github/com/presentation/core/dagger/component/AppComponent.kt
package homepunk.github.com.presentation.common.dagger.component

import android.app.Application
import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjectionModule
import dagger.android.support.AndroidSupportInjectionModule
import homepunk.github.com.presentation.VUApplication
import homepunk.github.com.presentation.common.dagger.module.*
import javax.inject.Singleton


/**Created by Homepunk on 26.12.2018. **/
@Component(modules = [
    AppModule::class,
    DataModule::class,
    ViewModelBindingModule::class,
    RepositoryBindingModule::class,
    FeatureBindingModule::class,
    AndroidSupportInjectionModule::class,
    AndroidInjectionModule::class
])
@Singleton
interface AppComponent {
    @Component.Builder
    interface Builder {
        @BindsInstance
        fun application(application: Application): Builder

        fun build(): AppComponent
    }

    fun inject(app: VUApplication)
}